addons/stock_account/models/stock_quant.py

```diff
@@ -93,7 +93,7 @@ def _get_inventory_move_values(self, qty, location_id, location_dest_id, package
         if not self.env.context.get('inventory_name'):
             force_period_date = self.env.context.get('force_period_date', False)
             if force_period_date:
-                res_move['name'] += _(' [Accounted on %s]', force_period_date)
+                res_move['inventory_name'] = res_move.get('inventory_name', '') + _(' [Accounted on %s]', force_period_date)
```

Comment:
Mm, after testing, I see that my suggestion isn't good. It needs to be something like:

```suggestion
                if self.product_uom_id.is_zero(qty):
                    name = _('Product Quantity Confirmed')
                else:
                    name = _('Product Quantity Updated')
                if self.env.uid and self.env.uid.id != SUPERUSER_ID:
                    name += f' ({self.env.uid.display_name})'
                res_move['inventory_name'] = name + _(' [Accounted on %s]', force_period_date)
```

`res_move['inventory_name'] = name + _(' [Accounted on %s]', force_period_date)`

Since as soon as we set an inventory_name, it won't do the `compute_inventory_name` anymore and it will lose that info. You'll need to import `SUPERUSER_ID` from Odoo and you should double check that this works as I didn't test it.